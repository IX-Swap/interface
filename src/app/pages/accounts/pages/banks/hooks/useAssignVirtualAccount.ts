import { getIdFromObj } from 'helpers/strings'
import { ValidCurrency } from 'helpers/types'
import { useAuth } from 'hooks/auth/useAuth'
import { useServices } from 'hooks/useServices'
import { useMutation, useQueryCache } from 'react-query'
import { virtualAccounts } from 'config/apiURL'
import { virtualAccountQueryKeys } from 'config/queryKeys'

export const useAssignVirtualAccount = (
  callback?: () => void,
  isAdditional: boolean | undefined = false
) => {
  const queryCache = useQueryCache()
  const { apiService, snackbarService } = useServices()
  const { user } = useAuth()
  const authUserId = getIdFromObj(user)
  const assignVirtualAccount = async (args: {
    currency: ValidCurrency
    userId?: string
  }) => {
    return await apiService.post(virtualAccounts.assign, {
      currency: args.currency,
      userId: args.userId ?? authUserId
    })
  }
  return useMutation(assignVirtualAccount, {
    onSuccess: async () => {
      callback?.()
      void snackbarService.showSnackbar(
        isAdditional
          ? 'Request to add app account sent'
          : 'Account has been assigned successfully!',
        'success'
      )
      await queryCache.invalidateQueries(virtualAccountQueryKeys.getByUserId)
    },
    onError: async (error: any) => {
      void snackbarService.showSnackbar(error.message, 'error')
      callback?.()
    }
  })
}
