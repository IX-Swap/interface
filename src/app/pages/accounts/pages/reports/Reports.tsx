import React from 'react'
import { Box, Grid, Typography } from '@mui/material'
import { PageHeader } from 'app/components/PageHeader/PageHeader'
import { VSpacer } from 'components/VSpacer'
import { Filters } from './components/Filters/Filters'
import { ReportsTable } from 'app/pages/accounts/pages/reports/components/ReportsTable/ReportsTable'

export const Reports: React.FC = () => {
  return (
    <Grid container direction={'column'}>
      <Grid item>
        <PageHeader title='Reports' />
      </Grid>
      <Grid item>
        <VSpacer size={'small'} />
        <Typography variant={'h4'}>Statements</Typography>
        <VSpacer size={'small'} />
      </Grid>
      <Grid
        item
        container
        justifyContent={'space-between'}
        alignItems={'center'}
      >
        <Grid item>
          <Box marginRight={2} marginTop={2} marginBottom={2}>
            <Typography variant={'body1'}>
              View, manage and track your accounting reports and taxations.
            </Typography>
          </Box>
        </Grid>
        <Grid item>
          <Filters />
        </Grid>
      </Grid>
      <Grid item>
        <VSpacer size={'medium'} />
        <ReportsTable />
      </Grid>
    </Grid>
  )
}
