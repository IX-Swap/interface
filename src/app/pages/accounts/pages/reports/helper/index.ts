import { formatAmount } from 'helpers/numbers'

export const formatValue = (value: string | number) =>
  typeof value === 'number' ? formatAmount(value) : value
