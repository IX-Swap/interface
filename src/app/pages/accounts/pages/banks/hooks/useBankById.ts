import { UseQueryData } from 'hooks/useParsedData'
import { useServices } from 'hooks/useServices'
import { useQuery } from 'react-query'
import { useAuth } from 'hooks/auth/useAuth'
import { Bank } from 'types/bank'
import { getIdFromObj, isEmptyString } from 'helpers/strings'
import { banksQueryKeys } from 'config/queryKeys'
import { accountsURL } from 'config/apiURL'

export interface UseBankByIdArgs {
  bankId?: string
  ownerId?: string
}

export const useBankById = (args: UseBankByIdArgs): UseQueryData<Bank> => {
  const { bankId, ownerId } = args
  const { apiService } = useServices()
  const { user } = useAuth()
  const userId = ownerId ?? getIdFromObj(user)
  const uri = accountsURL.banks.getById(userId, bankId)

  const getBank = async () => await apiService.get<Bank>(uri)
  const { data, ...rest } = useQuery(
    [banksQueryKeys.getById, userId, bankId],
    getBank,
    { enabled: !isEmptyString(bankId) && !isEmptyString(userId) }
  )

  return {
    ...rest,
    data: data?.data
  }
}
