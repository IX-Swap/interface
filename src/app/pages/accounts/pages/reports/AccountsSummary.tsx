import React from 'react'
import { Grid } from '@material-ui/core'
import { PageHeader } from 'app/components/PageHeader/PageHeader'
import { VSpacer } from 'components/VSpacer'
import { ReportsInfo } from 'app/pages/accounts/pages/reports/components/ReportsInfo/ReportsInfo'
import { Actions } from './components/Actions/Actions'
import { ReportsAccordion } from 'app/pages/accounts/pages/reports/components/ReportsAccordion/ReportsAccordion'
import { useGetActivitySummary } from 'app/pages/accounts/hooks/useGetActivitySummary'

export const AccountsSummary: React.FC = () => {
  const { data, isLoading } = useGetActivitySummary()
  console.log(data, isLoading)

  return (
    <Grid container direction={'column'}>
      <Grid item>
        <PageHeader title='Activity Summary' />
        <VSpacer size={'medium'} />
      </Grid>

      <ReportsInfo>
        <Actions onContractClick={() => null} onExpandClick={() => null} />
      </ReportsInfo>

      <Grid item>
        <VSpacer size={'medium'} />

        <ReportsAccordion summary={'Open Positions'}>
          <div>Open Positions</div>
        </ReportsAccordion>

        <ReportsAccordion summary={'Cash Report'}>
          <div>Cash Report</div>
        </ReportsAccordion>
      </Grid>
    </Grid>
  )
}
