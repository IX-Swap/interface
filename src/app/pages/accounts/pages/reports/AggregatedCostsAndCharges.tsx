import React, { useEffect } from 'react'
import { Grid } from '@material-ui/core'
import { PageHeader } from 'app/components/PageHeader/PageHeader'
import { VSpacer } from 'components/VSpacer'
import { ReportsInfo } from 'app/pages/accounts/pages/reports/components/ReportsInfo/ReportsInfo'
import { ReportsAccordion } from 'app/pages/accounts/pages/reports/components/ReportsAccordion/ReportsAccordion'
import { useFeeAndCharges } from 'app/pages/accounts/hooks/useFeeAndCharges'
import { FeesTable } from 'app/pages/accounts/pages/reports/components/FeesTable/FeesTable'
import { LoadingIndicator } from 'app/components/LoadingIndicator/LoadingIndicator'
import { NoData } from 'app/pages/accounts/pages/reports/components/NoData/NoData'
import { useQueryFilter } from 'hooks/filters/useQueryFilter'

export const AggregatedCostsAndCharges: React.FC = () => {
  const { data, isLoading } = useFeeAndCharges()
  const { updateFilter } = useQueryFilter()

  useEffect(() => {
    updateFilter('expandedSections', 'Fees (SGD),Fees (USD)')
  }, [])

  const hasSGDData = !!data?.sgd?.length
  const hasUSDData = !!data?.usd?.length

  const renderContent = () => {
    if (!hasSGDData && !hasUSDData) {
      return <NoData />
    }
    if (hasSGDData) {
      return (
        <ReportsAccordion summary={'Fees (SGD)'}>
          <FeesTable accounts={data.sgd} total={data.totalSgd} />
        </ReportsAccordion>
      )
    }

    if (hasUSDData) {
      return (
        <ReportsAccordion summary={'Fees (USD)'}>
          <FeesTable accounts={data.usd} total={data.totalUsd} />
        </ReportsAccordion>
      )
    }
  }

  if (isLoading) {
    return <LoadingIndicator />
  }

  return (
    <Grid container direction={'column'}>
      <Grid item>
        <PageHeader title='Aggregated Costs and Charges' />
        <VSpacer size={'medium'} />
      </Grid>

      <ReportsInfo />

      <Grid item>
        <VSpacer size={'medium'} />
        {renderContent()}
      </Grid>
    </Grid>
  )
}
