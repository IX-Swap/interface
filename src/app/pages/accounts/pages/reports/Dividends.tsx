import React, { useEffect } from 'react'
import { Grid } from '@mui/material'
import { PageHeader } from 'app/components/PageHeader/PageHeader'
import { VSpacer } from 'components/VSpacer'
import { ReportsInfo } from 'app/pages/accounts/pages/reports/components/ReportsInfo/ReportsInfo'
import { ReportsAccordion } from 'app/pages/accounts/pages/reports/components/ReportsAccordion/ReportsAccordion'
import { useDividends } from 'app/pages/accounts/hooks/useDividends'
import { DividendsTable } from 'app/pages/accounts/pages/reports/components/DividendsTable/DividendsTable'
import { LoadingIndicator } from 'app/components/LoadingIndicator/LoadingIndicator'
import { NoData } from 'app/pages/accounts/pages/reports/components/NoData/NoData'
import { useQueryFilter } from 'hooks/filters/useQueryFilter'

export const Dividends: React.FC = () => {
  const { updateFilter } = useQueryFilter()
  const { data, isLoading } = useDividends()

  useEffect(() => {
    updateFilter('expandedSections', 'Dividends')
  }, [])

  const renderContent = () => {
    if (data === undefined || data.length < 1) {
      return <NoData />
    }
    return (
      <ReportsAccordion summary={'Dividends'}>
        <DividendsTable data={data} />
      </ReportsAccordion>
    )
  }

  if (isLoading) {
    return <LoadingIndicator />
  }

  return (
    <Grid container direction={'column'}>
      <Grid item>
        <PageHeader title='Dividends' />
        <VSpacer size={'medium'} />
      </Grid>

      <ReportsInfo />

      <Grid item>
        <VSpacer size={'medium'} />
        {renderContent()}
      </Grid>
    </Grid>
  )
}
