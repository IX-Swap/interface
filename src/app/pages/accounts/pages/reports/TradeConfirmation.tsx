import React from 'react'
import { Grid } from '@material-ui/core'
import { PageHeader } from 'app/components/PageHeader/PageHeader'
import { VSpacer } from 'components/VSpacer'
import { ReportsInfo } from 'app/pages/accounts/pages/reports/components/ReportsInfo/ReportsInfo'
import { ReportsAccordion } from 'app/pages/accounts/pages/reports/components/ReportsAccordion/ReportsAccordion'
import { useGetExchangeFills } from 'app/pages/accounts/hooks/useGetExchangeFills'

export const TradeConfirmation: React.FC = () => {
  const { data, isLoading } = useGetExchangeFills()
  console.log(data, isLoading)

  return (
    <Grid container direction={'column'}>
      <Grid item>
        <PageHeader title='Trade Confirmation' />
        <VSpacer size={'medium'} />
      </Grid>

      <ReportsInfo />

      <Grid item>
        <VSpacer size={'medium'} />

        <ReportsAccordion summary={'Trades'}>
          <div>Trades</div>
        </ReportsAccordion>
      </Grid>
    </Grid>
  )
}
