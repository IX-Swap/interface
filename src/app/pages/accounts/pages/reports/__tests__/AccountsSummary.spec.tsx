import React from 'react'
import { render } from 'test-utils'
import { AccountsSummary } from 'app/pages/accounts/pages/reports/AccountsSummary'
import * as useActivitySummary from 'app/pages/accounts/hooks/useActivitySummary'
import { fakeActivitySummary } from '__fixtures__/reports'
import { generateQueryResult } from '__fixtures__/useQuery'

describe('AccountsSummary', () => {
  it('should match snapshot when data is undefined', () => {
    jest.spyOn(useActivitySummary, 'useActivitySummary').mockReturnValue(
      generateQueryResult({
        data: undefined,
        isLoading: false
      })
    )

    const { container } = render(<AccountsSummary />)
    expect(container).toMatchSnapshot()
  })

  it('should match snapshot when all data is loaded successfully', () => {
    jest.spyOn(useActivitySummary, 'useActivitySummary').mockReturnValue(
      generateQueryResult({
        data: fakeActivitySummary,
        isLoading: false
      })
    )

    const { container } = render(<AccountsSummary />)
    expect(container).toMatchSnapshot()
  })

  it('should match snapshot when data is loaded successfully and openPositions is empty array', () => {
    jest.spyOn(useActivitySummary, 'useActivitySummary').mockReturnValue(
      generateQueryResult({
        data: { fakeActivitySummary, openPositions: [] },
        isLoading: false
      })
    )

    const { container } = render(<AccountsSummary />)
    expect(container).toMatchSnapshot()
  })

  it('should match snapshot when data is loaded successfully and fakeCashReports is empty array', () => {
    jest.spyOn(useActivitySummary, 'useActivitySummary').mockReturnValue(
      generateQueryResult({
        data: { fakeActivitySummary, cashReports: [] },
        isLoading: false
      })
    )

    const { container } = render(<AccountsSummary />)
    expect(container).toMatchSnapshot()
  })
})
