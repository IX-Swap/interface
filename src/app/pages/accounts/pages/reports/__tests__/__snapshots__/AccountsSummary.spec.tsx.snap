// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountsSummary should match snapshot when all data is loaded successfully 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column"
  >
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <div
        class="MuiGrid-root makeStyles-container-22 MuiGrid-container MuiGrid-direction-xs-column"
      >
        <div
          class="MuiGrid-root makeStyles-header-21 MuiGrid-container"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center"
          >
            <h2
              class="MuiTypography-root MuiTypography-h2"
            >
              Activity Summary
            </h2>
          </div>
        </div>
        <div
          class="MuiGrid-root"
        >
          <nav
            class="MuiTypography-root MuiBreadcrumbs-root MuiTypography-body1 MuiTypography-colorTextSecondary"
          >
            <ol
              class="MuiBreadcrumbs-ol"
            />
          </nav>
        </div>
      </div>
      <div
        class="MuiBox-root MuiBox-root-24"
      />
    </div>
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <div
        class="MuiBox-root MuiBox-root-28"
      />
      <div
        class="MuiPaper-root MuiAccordion-root makeStyles-root-29 MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          aria-disabled="false"
          aria-expanded="false"
          class="MuiButtonBase-root MuiAccordionSummary-root makeStyles-summaryWrapper-30"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiAccordionSummary-content makeStyles-summaryContent-31"
          >
            <p
              class="MuiTypography-root makeStyles-summary-32 MuiTypography-body1 MuiTypography-colorPrimary"
            >
              Open Positions
            </p>
          </div>
          <div
            aria-disabled="false"
            aria-hidden="true"
            class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-edgeEnd"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-colorPrimary"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
        </div>
        <div
          class="MuiCollapse-root MuiCollapse-hidden"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner"
            >
              <div
                role="region"
              >
                <div
                  class="MuiAccordionDetails-root makeStyles-detailsWrapper-33"
                >
                  <div
                    class="MuiTableContainer-root"
                  >
                    <table
                      class="MuiTable-root"
                    >
                      <thead
                        class="MuiTableHead-root"
                      >
                        <tr
                          class="MuiTableRow-root MuiTableRow-head"
                        >
                          <th
                            class="MuiTableCell-root MuiTableCell-head makeStyles-headColumn-34 MuiTableCell-alignLeft"
                            scope="col"
                          >
                            Symbol
                          </th>
                          <th
                            class="MuiTableCell-root MuiTableCell-head makeStyles-headColumn-34 MuiTableCell-alignLeft"
                            scope="col"
                          >
                            Open
                          </th>
                          <th
                            class="MuiTableCell-root MuiTableCell-head makeStyles-headColumn-34 MuiTableCell-alignRight"
                            scope="col"
                          >
                            Quantity
                          </th>
                          <th
                            class="MuiTableCell-root MuiTableCell-head makeStyles-headColumn-34 MuiTableCell-alignRight"
                            scope="col"
                          >
                            Cost price
                          </th>
                          <th
                            class="MuiTableCell-root MuiTableCell-head makeStyles-headColumn-34 MuiTableCell-alignRight"
                            scope="col"
                          >
                            Cost value
                          </th>
                          <th
                            class="MuiTableCell-root MuiTableCell-head makeStyles-headColumn-34 MuiTableCell-alignRight"
                            scope="col"
                          >
                            Last Trade Price
                          </th>
                          <th
                            class="MuiTableCell-root MuiTableCell-head makeStyles-headColumn-34 MuiTableCell-alignRight"
                            scope="col"
                          >
                            Current Value
                          </th>
                          <th
                            class="MuiTableCell-root MuiTableCell-head makeStyles-headColumn-34 MuiTableCell-alignRight"
                            scope="col"
                          >
                            Unrealized P/L
                          </th>
                        </tr>
                      </thead>
                      <tbody
                        class="MuiTableBody-root"
                      >
                        <tr
                          class="MuiTableRow-root makeStyles-row-39 makeStyles-row-44"
                        >
                          <th
                            class="MuiTableCell-root MuiTableCell-body makeStyles-column-36"
                            role="cell"
                            scope="row"
                          >
                            Tokens
                          </th>
                          <td
                            class="MuiTableCell-root MuiTableCell-body makeStyles-column-36 MuiTableCell-alignRight"
                          />
                          <td
                            class="MuiTableCell-root MuiTableCell-body makeStyles-column-36 MuiTableCell-alignRight"
                          />
                          <td
                            class="MuiTableCell-root MuiTableCell-body makeStyles-column-36 MuiTableCell-alignRight"
                          />
                          <td
                            class="MuiTableCell-root MuiTableCell-body makeStyles-column-36 MuiTableCell-alignRight"
                          />
                          <td
                            class="MuiTableCell-root MuiTableCell-body makeStyles-column-36 MuiTableCell-alignRight"
                          />
                          <td
                            class="MuiTableCell-root MuiTableCell-body makeStyles-column-36 MuiTableCell-alignRight"
                          />
                          <td
                            class="MuiTableCell-root MuiTableCell-body makeStyles-column-36 MuiTableCell-alignRight"
                          />
                        </tr>
                        <tr
                          class="MuiTableRow-root makeStyles-row-39 makeStyles-row-46"
                        >
                          <th
                            class="MuiTableCell-root MuiTableCell-body makeStyles-column-36"
                            role="cell"
                            scope="row"
                          >
                            RHTC/SGD
                          </th>
                          <td
                            class="MuiTableCell-root MuiTableCell-body makeStyles-column-36 MuiTableCell-alignRight"
                          >
                            2021-11-01, 06:15 AM
                          </td>
                          <td
                            class="MuiTableCell-root MuiTableCell-body makeStyles-column-36 MuiTableCell-alignRight"
                          >
                            3
                          </td>
                          <td
                            class="MuiTableCell-root MuiTableCell-body makeStyles-column-36 MuiTableCell-alignRight"
                          >
                            2,000.00
                          </td>
                          <td
                            class="MuiTableCell-root MuiTableCell-body makeStyles-column-36 MuiTableCell-alignRight"
                          >
                            6,000.00
                          </td>
                          <td
                            class="MuiTableCell-root MuiTableCell-body makeStyles-column-36 MuiTableCell-alignRight"
                          >
                            2,000.00
                          </td>
                          <td
                            class="MuiTableCell-root MuiTableCell-body makeStyles-column-36 MuiTableCell-alignRight"
                          >
                            6,000.00
                          </td>
                          <td
                            class="MuiTableCell-root MuiTableCell-body makeStyles-column-36 MuiTableCell-alignRight"
                          >
                            0.00
                          </td>
                        </tr>
                        <tr
                          class="MuiTableRow-root makeStyles-row-39 makeStyles-row-48"
                        >
                          <th
                            class="MuiTableCell-root MuiTableCell-body makeStyles-column-36"
                            role="cell"
                            scope="row"
                          >
                            Total
                          </th>
                          <td
                            class="MuiTableCell-root MuiTableCell-body makeStyles-column-36 MuiTableCell-alignRight"
                          />
                          <td
                            class="MuiTableCell-root MuiTableCell-body makeStyles-column-36 MuiTableCell-alignRight"
                          />
                          <td
                            class="MuiTableCell-root MuiTableCell-body makeStyles-column-36 MuiTableCell-alignRight"
                          >
                            5,033.00
                          </td>
                          <td
                            class="MuiTableCell-root MuiTableCell-body makeStyles-column-36 MuiTableCell-alignRight"
                          />
                          <td
                            class="MuiTableCell-root MuiTableCell-body makeStyles-column-36 MuiTableCell-alignRight"
                          />
                          <td
                            class="MuiTableCell-root MuiTableCell-body makeStyles-column-36 MuiTableCell-alignRight"
                          >
                            146,000.00
                          </td>
                          <td
                            class="MuiTableCell-root MuiTableCell-body makeStyles-column-36 MuiTableCell-alignRight"
                          >
                            -79,835.00
                          </td>
                        </tr>
                      </tbody>
                    </table>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`AccountsSummary should match snapshot when data is loaded successfully and fakeCashReports is empty array 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column"
  >
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <div
        class="MuiGrid-root makeStyles-container-73 MuiGrid-container MuiGrid-direction-xs-column"
      >
        <div
          class="MuiGrid-root makeStyles-header-72 MuiGrid-container"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center"
          >
            <h2
              class="MuiTypography-root MuiTypography-h2"
            >
              Activity Summary
            </h2>
          </div>
        </div>
        <div
          class="MuiGrid-root"
        >
          <nav
            class="MuiTypography-root MuiBreadcrumbs-root MuiTypography-body1 MuiTypography-colorTextSecondary"
          >
            <ol
              class="MuiBreadcrumbs-ol"
            />
          </nav>
        </div>
      </div>
      <div
        class="MuiBox-root MuiBox-root-75"
      />
    </div>
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <div
        class="MuiBox-root MuiBox-root-79"
      />
      <div
        class="MuiPaper-root MuiAccordion-root makeStyles-root-80 MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          aria-disabled="false"
          aria-expanded="false"
          class="MuiButtonBase-root MuiAccordionSummary-root makeStyles-summaryWrapper-81"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiAccordionSummary-content makeStyles-summaryContent-82"
          >
            <p
              class="MuiTypography-root makeStyles-summary-83 MuiTypography-body1 MuiTypography-colorPrimary"
            >
              Cash Report
            </p>
          </div>
          <div
            aria-disabled="false"
            aria-hidden="true"
            class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-edgeEnd"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-colorPrimary"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
        </div>
        <div
          class="MuiCollapse-root MuiCollapse-hidden"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner"
            >
              <div
                role="region"
              >
                <div
                  class="MuiAccordionDetails-root makeStyles-detailsWrapper-84"
                >
                  <div
                    class="MuiTableContainer-root"
                  >
                    <table
                      class="MuiTable-root"
                    >
                      <tbody
                        class="MuiTableBody-root"
                      />
                    </table>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`AccountsSummary should match snapshot when data is loaded successfully and openPositions is empty array 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column"
  >
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <div
        class="MuiGrid-root makeStyles-container-54 MuiGrid-container MuiGrid-direction-xs-column"
      >
        <div
          class="MuiGrid-root makeStyles-header-53 MuiGrid-container"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center"
          >
            <h2
              class="MuiTypography-root MuiTypography-h2"
            >
              Activity Summary
            </h2>
          </div>
        </div>
        <div
          class="MuiGrid-root"
        >
          <nav
            class="MuiTypography-root MuiBreadcrumbs-root MuiTypography-body1 MuiTypography-colorTextSecondary"
          >
            <ol
              class="MuiBreadcrumbs-ol"
            />
          </nav>
        </div>
      </div>
      <div
        class="MuiBox-root MuiBox-root-56"
      />
    </div>
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <div
        class="MuiBox-root MuiBox-root-60"
      />
      <div
        class="MuiGrid-root makeStyles-wrapper-61 MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center"
      >
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <div
            class="MuiBox-root MuiBox-root-64"
          />
          <svg
            style="max-width: 100%;"
          >
            no_data.svg
          </svg>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <div
            class="MuiBox-root MuiBox-root-65"
          />
          <p
            class="MuiTypography-root makeStyles-subtitle-62 MuiTypography-body1"
          >
            This page is empty.
          </p>
          <div
            class="MuiBox-root MuiBox-root-66"
          />
          <p
            class="MuiTypography-root makeStyles-content-63 MuiTypography-body1"
          >
            There are no records to display for the selected report.
          </p>
          <div
            class="MuiBox-root MuiBox-root-67"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`AccountsSummary should match snapshot when data is undefined 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column"
  >
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <div
        class="MuiGrid-root makeStyles-container-3 MuiGrid-container MuiGrid-direction-xs-column"
      >
        <div
          class="MuiGrid-root makeStyles-header-2 MuiGrid-container"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center"
          >
            <h2
              class="MuiTypography-root MuiTypography-h2"
            >
              Activity Summary
            </h2>
          </div>
        </div>
        <div
          class="MuiGrid-root"
        >
          <nav
            class="MuiTypography-root MuiBreadcrumbs-root MuiTypography-body1 MuiTypography-colorTextSecondary"
          >
            <ol
              class="MuiBreadcrumbs-ol"
            />
          </nav>
        </div>
      </div>
      <div
        class="MuiBox-root MuiBox-root-5"
      />
    </div>
    <div
      class="MuiGrid-root makeStyles-wrapper-9 MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-center"
      style="opacity: 0.75;"
    >
      <div
        class="MuiBox-root MuiBox-root-10"
      />
      <div
        class="MuiBox-root MuiBox-root-11"
      />
      <div
        class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
        role="progressbar"
        style="width: 40px; height: 40px;"
      >
        <svg
          class="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <div
        class="MuiBox-root MuiBox-root-12"
      />
      <div
        class="MuiGrid-root makeStyles-wrapper-13 MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center"
      >
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <div
            class="MuiBox-root MuiBox-root-16"
          />
          <svg
            style="max-width: 100%;"
          >
            no_data.svg
          </svg>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <div
            class="MuiBox-root MuiBox-root-17"
          />
          <p
            class="MuiTypography-root makeStyles-subtitle-14 MuiTypography-body1"
          >
            This page is empty.
          </p>
          <div
            class="MuiBox-root MuiBox-root-18"
          />
          <p
            class="MuiTypography-root makeStyles-content-15 MuiTypography-body1"
          >
            There are no records to display for the selected report.
          </p>
          <div
            class="MuiBox-root MuiBox-root-19"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
